src/reducers/messageReducer.js

```diff
@@ -1,11 +1,18 @@
 import i18n from '@dhis2/d2-i18n'
 import log from 'loglevel'
-import * as actions from '../constants/actions'
-import messageTypes from '../constants/messageTypes'
-import { POSITIVE, NEGATIVE, NEUTRAL } from '../constants/development'
-import { findIndexOfId } from '../utils/helpers'
+import * as actions from '../constants/actions.js'
+import messageTypes from '../constants/messageTypes.js'
+import { POSITIVE, NEGATIVE, NEUTRAL } from '../constants/development.js'
+import { findIndexOfId } from '../utils/helpers.js'
 
 export const initialState = {
```

Comment:
Since this reducer is already so big, and the currentUser state updates seem to be completely separate from the other state, you could consider using a separate reducer for the currentUser state. That might make things more readable/easier to parse.

Reply:
Yeah probably, I was just sticking to the approach that was in place already and didn't want to start refactoring the state design too much... This also related to what I said in https://github.com/dhis2/messaging-app/pull/230#discussion_r516629388
> At some point we need to refactor this app to use declarative data fetching and at that stage we'll probably get rid of redux all together.

I can update this I guess, I'll add a separate reducer for it.

Reply:
After diving into this a bit, I think I'd rather leave it the way it is. The messaging state already contains several unrelated/standalone properties. Putting the currentUser in a separate reducer but keeping the rest where it is will only add to the confusion.

Reply:
Ok. I think that we should prioritize addressing the tech debt for this app after this PR. The structure is basically preventing us from following normal best practices (not your doing, but thought I'd mention it).

Reply:
yes, completely agree

Reply:
Nice, sounds good 👍